main/provider/index.js

```diff
@@ -321,21 +321,28 @@ class Provider extends EventEmitter {
     }, targetChain)
   }
 
-  sendTransaction (payload, res) {
-    const rawTx = this.getRawTx(payload)
-    const activeAccount = accounts.current()
+  fillTransaction (rawTx, cb) {
+    delete rawTx.gas
+    delete rawTx.gasLimit
+    
     const gas = this._gasFees(rawTx)
     const chainConfig = this.connection.connections['ethereum'][parseInt(rawTx.chainId)].chainConfig
 
-    const estimateGas = rawTx.gasLimit
-      ? Promise.resolve(rawTx)
-      : this._getGasEstimate(rawTx)
-        .then(gasLimit => ({ ...rawTx, gasLimit }))
-        .catch(err => ({ ...rawTx, gasLimit: '0x00', warning: err.message }))
-
-    estimateGas
+    this._getGasEstimate(rawTx)
+      .then(gasLimit => ({ ...rawTx, gasLimit }))
       .then(tx => populateTransaction(tx, chainConfig, gas))
-      .then(tx => {
+      .then(tx => cb(null, tx))
+      .catch(err => cb(null, ({ ...rawTx, gasLimit: '0x00', warning: err.message })))
```

Comment:
if there's an error calculating the gas limit then we will never populate the gas prices. is this intentional?